sdk/resourcemanager/Azure.ResourceManager/src/ArmClient.cs

```diff
@@ -111,15 +110,43 @@ internal virtual async Task<bool> CanUseTagResourceAsync(CancellationToken cance
 
         private void CopyApiVersionOverrides(ArmClientOptions options)
         {
+            if (options.ApiVersionProfile != null)
+            {
+                // Refer to https://github.com/Azure/azure-rest-api-specs/tree/main/profile#file-structure for data structure.
+                var allProfile = options.ApiVersionProfile.ToObjectFromJson() as Dictionary<string, object>;
+                var armProfile = allProfile["resource-manager"] as Dictionary<string, object>;
+                foreach (var keyValuePair in armProfile)
```

Comment:
could we change this in this way so that this is more readable and simple?
```suggestion
                foreach ((string namespaceName, Dictionary<string, object> profile) in armProfile)
```
I this way I believe we do not need the cast

Reply:
Unfortunately, the .NET Standard 2.0 framework the SDK uses [does not support Deconstruct](https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.keyvaluepair-2.deconstruct?view=net-7.0#applies-to) method of `KeyValuePairs`.